doc/system-telemetry/process-docker-stats.md

```diff
@@ -0,0 +1,84 @@
+# Process and docker stats availability thorugh telemetry agent
+
+## Revision
+
+| Rev | Date     | Author      | Change Description |
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 09/12/19 | Pradnya Mohite | Initial version    |
+
+## Scope
+-Enable sonic streaming telemetry agent to send Process and docker stats data
+
+### Enable sonic streaming telemetry agent to send Process and docker stats data
+
+##### Part 1 
+For 1st part, Daemon code will be added under sonic-buildimage/files/image_config.  A Daemon will start when OS starts and upload Process and docker stats data every 2 mins to state-DB. Details of CLI and state-DB given below. 
+
+##### Part 2
+From state-DB data need to be available via telemetry agent
+
+##### CLI output  and corresponding structure in state-DB for process and docket stats
+
+###### Process stats
+
+$ ps aux 
+
+|USER |   PID |  %CPU|%MEM |VSZ          |RSS   |  TTY |   STAT |START| TIME| COMMAND                                                                                                                                                                                                                                                                                                                                                                                                                                               |
+|-------|-------|--------|--------|------------|-------|------|---------|-------|------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+|root    | 4276 | 0.0    | 0.0     | 108816    | 5552 |?      |    Sl     |00:39 |0:01 | containerd-shim -namespace moby -workdir /var/lib/containerd/io.containerd.runtime.v1.linux/moby/07983d8d914904ac8054af2be0aa6aa70a8325700aa2588f7424ece3fbfe648c -address /run/containerd/containerd.sock -containerd-binary /usr/bin/containerd -runtime-root /var/run/docker/runtime-runc|
+|root    | 6601 |  0.0   |  0.0    |108816     | 5516 |?      |    Sl     |00:42 |0:01 |containerd-shim -namespace moby -workdir /var/lib/containerd/io.containerd.runtime.v1.linux/moby/4dc60c74334813d6c833d967b1196d1783b90bff0488aa0c35d544db66dc8a81 -address /run/containerd/containerd.sock -containerd-binary /usr/bin/containerd -runtime-root /var/run/docker/runtime-runc|
+
+above output will be stored inside state-DB as follows:
+
+ProcessStats|4276
```

Comment:
process id can change, the program needs to remove non existent process id from the state db.

Reply:
What way do you suggest? One way can be setting key expiry  time in state_db?

Reply:
If the daemon doesn't crash, then after it scrapes the latest data and before it uploads it to the DB, it can do a diff of the current DB contents and delete entries which no longer exist.

However, if the daemon were to ever crash, stale data could be present in the DB, and would be sent out via streaming telemetry. Maybe for this case it would be good to include a timestamp stating when the data was collected. This way any services dependent on this data can decide whether or not to use the data based on the age of the timestamp.

Reply:
the monitor process should be restarted if it crashes, otherwise we will not have process data updated in the database. for the timestamp, maybe we do not need to add for every entry since they are all updated at the same time.

Reply:
Agree it *should* be restarted, but just in case...

Also agree that one timestamp is sufficient.

Reply:
Daemon on start will delete all entries from state db and then update. This way non existent process id won't persist in db. Adding new entry as "Process_stats|LastUpdateTime" to track timestamp. Added this in functional spec part1.